iwyu.cc

```diff
@@ -2594,8 +2583,17 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     vector<const UsingDecl*> using_decls
         = FindInMultiMap(visitor_state_->using_declarations, decl);
     for (Each<const UsingDecl*> it(&using_decls); !it.AtEnd(); ++it) {
-      if (!(*it)->getDeclContext()->Encloses(using_context))
-        continue;
+      const DeclContext* using_decl_context = (*it)->getDeclContext();
+      if (!using_decl_context->Encloses(use_context)) {
```

Comment:
You said in the original issue:
> This fix is not small and contains a questionable check at line 2577 of iwyu.cc that I think warrants some discussion.

Is this the one? I can't quite make out what this means -- can you say something about what this code block does, in plain English?